haystack/nodes/other/shaper.py

```diff
@@ -71,7 +92,18 @@ def invoke(self, parent, node: Dict[str, Any], invocation_context: Dict[str, Any
                     f"an assignment. The input configuration is {node}"
                 )
 
-    def invoke_function(self, parent, node, invocation_context: Dict[str, Any]):
+    def invoke_function(self, parent: str, node: Dict[str, Any], invocation_context: Dict[str, Any]):
+        """
+        Invoke a function on the given invocation context and stores the result back in the invocation context.
+
+        :param parent: The parent node of the current node.
+        :type parent: str
```

Comment:
Same here, I don't think we need the types since we extract them from the function definition.